src/hypofuzz/database.py

```diff
@@ -36,6 +37,7 @@ class Report(TypedDict):
 
 
 class Metadata(TypedDict):
+    database_key: str
     nodeid: str
     seed_pool: list[list[str]]
     failures: list[list[str]]
```

Comment:
This is a breaking change for existing databases. We should work out how to do migrations at some point, but I think we have a while yet before we stabilize the db to a point where that becomes worthwhile

(I'm using `database_key` as the unique key for react's rendering, and in general we've discussed before that `database_key` should be the backing unique key while node id is for display only)

Reply:
See https://github.com/Zac-HD/hypofuzz/issues/3#issuecomment-2808734642

- We can, in general, safely drop basically anything from the database except for our saved inputs.  It costs us pretty graphs and some state estimation which makes the fuzzer less efficient for... minutes to hours (?), but it's recoverable.  
  - We should test that we're OK if the database has unexpected data shapes in it though; and "parse don't validate" suggests loading as frozen dataclasses rather than dicts - that's a natural place to put migration logic to make new objects out of older saved data.
- there are enough database changes in that comment that I suggest we do them first, then rebase these frontend PRs.  Sorry!  Note that we can just show the latest run for now, and do aggregation later.